pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -0,0 +1,158 @@
+                                </NavItem>
+                                {this.props.ansible &&
+                                    <NavItem eventKey={2}>
+                                        {_("Ansible script")}
+                                    </NavItem>
+                                }
+                            </Nav>
+                            <TabContent animation>
+                                <TabPane eventKey={1}>
+                                    <pre>
+                                        {this.props.shell}
+                                    </pre>
+                                </TabPane>
+                                {this.props.ansible &&
+                                    <TabPane eventKey={2}>
+                                        {this.props.ansible}
+                                    </TabPane>
+                                }
+                            </TabContent>
+                        </React.Fragment>
+                    </TabContainer>
+                </Modal.Body>
+                <Modal.Footer>
+                    <Button bsStyle='default' className='btn-cancel' onClick={this.props.onClose}>
+                        { _("Close") }
+                    </Button>
+                </Modal.Footer>
+            </Modal>
+        );
+    }
+}
+
+ModificationsExportDialog.propTypes = {
+    shell: PropTypes.string.isRequired,
+    ansible: PropTypes.string,
+    show: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+};
+
+/* Display list of modification in human readable format
+ *
+ * Also show `View automation script` button which opens dialog in which different
+ * scripts are available. With these script it is possible to get other macchines
+ * to the same configuration.
+ *
+ * Pass array `entries` to show human readable messages.
+ * Pass string `shell` and `ansible` with scripts.
+ *
+ */
+export class Modifications extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showDialog: false,
+        };
+    }
+
+    render() {
+        var emptyRow = null;
+        if (!this.props.entries || this.props.entries.length === 0) {
+            var markup = {
+                __html: this.props.emptyCaption,
+            };
+            emptyRow = <thead className={"listing-ct-empty"}>
+                <tr className="modification-row"><td dangerouslySetInnerHTML={ markup } /></tr>
+            </thead>;
+        }
+
+        return (
+            <React.Fragment>
+                <ModificationsExportDialog show={this.state.showDialog} shell={ this.props.shell } ansible={ this.props.ansible } onClose={ () => this.setState({ showDialog: false }) } />
+                <table className={ "listing-ct modifications-table" }>
+                    <caption className="cockpit-caption">
+                        <div className="modifications-caption">
+                            { this.props.title }
```

Comment:
Why isn't this using a standard `<h3>`? Probably more a11y and we could lose the custom CSS?

Reply:
Because it is in `<caption>` element, size of the title is set up through `listing.less` style. Purpose of this is to be able to show the "View automation script" in the same line and to make it smaller than the caption itself.